### ISSUE TITLE
[Feature Request] Support for "Install macOS ...." img/iso (on Intel Macs)

### ISSUE THREAD

**Issue by TheAlienDrew** [ventoy/Ventoy#918]
## To be clear here, I'm not asking for support in making hackintosh images bootable. I'm asking for normal unmodified apple usb install images (e.g. "Install macOS Big Sur") to be bootable from ventoy.

If you don't know what I mean about these bootable usb install images, similar to making a bootable USB (on a Mac device), you can also create bootable images that could then be imaged to a USB drive.

Below is an example of creating such an image file (assuming that the "Install macOS Big Sur.app" is located in "/Applications":
```sh
hdiutil create -o /tmp/Install\ macOS\ Big\ Sur -size 16000m -layout SPUD -fs HFS+J
hdiutil attach /tmp/Install\ macOS\ Big\ Sur.dmg -noverify -mountpoint /Volumes/Install\ macOS\ Big\ Sur
sudo /Applications/Install\ macOS\ Big\ Sur.app/Contents/Resources/createinstallmedia --volume /Volumes/Install\ macOS\ Big\ Sur
```
... then open Disk Utility and unmount any "Shared Support" volumes, then continue below (only happens when dealing with Big Sur; Catalina and Mojave install apps aren't affected by this)...
```sh
hdiutil detach /Volumes/Install\ macOS\ Big\ Sur
mv /tmp/Install\ macOS\ Big\ Sur.dmg ~/Desktop/
... however, the dmg file is technically a raw image file anyways, so we can just rename that to "Install macOS Big Sur.img"

Optionally, though shouldn't be required, you can convert the dmg to an iso ...
```sh
hdiutil convert ~/Desktop/Install\ macOS\ Big\ Sur.dmg -format UDTO -o ~/Desktop/Install\ macOS\ Big\ Sur
mv ~/Desktop/Install\ macOS\ Big\ Sur.cdr ~/Desktop/Install\ macOS\ Big\ Sur.iso
```

Anyways, the resulting image should be bootable if it were to be imaged to a USB device.

However, trying to use them with Ventoy doesn't work, from what I assume is likely because Ventoy doesn't support the boot process the way macs read and boot from a normally made install drive.

I don't know much of anything about bootloaders, but I did notice "/System/Library/CoreServices/boot.efi" from the root of any install image (along with other directories and folders that appear related) that could assist in getting them to boot correctly?

If not, there's always [Clover bootloader source code](https://github.com/CloverHackyColor/CloverBootloader) to look at to maybe boot them correctly?

**Comment by TheAlienDrew** [comment]
Maybe looking at how VirtualBox works to boot the images may help? https://github.com/mirror/vbox/tree/master/src/VBox/Devices/

**Comment by TheAlienDrew** [comment]
Looking a little at CloverBootloader source code, I can confirm that `/System/Library/CoreServices/boot.efi` is the right boot file and it should be UEFI... @ventoy you just need to somehow get that loaded in, if I'm not mistaken.

**Comment by TheAlienDrew** [comment]
Researching a bit further, found https://www.geekrar.com/fix-efi-internal-shell-on-macos-mojave-on-virtualbox/

And now if I understand correctly, the biggest issue that needs to be overcome, is having HFS+ filesystem drivers that boot with Ventoy so that it can open the Install macOS dmg/img files, and then boot from the boot.efi file in `/System/Library/CoreServices/boot.efi`

**Comment by TheAlienDrew** [comment]
@ventoy and actually it does look like VirtualBox source code could help:
- https://github.com/mirror/vbox/blob/master/src/VBox/Devices/EFI/Firmware/VBoxPkg/VBoxFsDxe/fsw_hfs.h
- https://github.com/mirror/vbox/blob/master/src/VBox/Devices/EFI/Firmware/VBoxPkg/VBoxFsDxe/fsw_hfs.c
- https://github.com/mirror/vbox/blob/master/src/VBox/Devices/EFI/Firmware/VBoxPkg/VBoxFsDxe/VBoxHfs.inf

Those are the HFS+ references I can find in the source code that should aid in creating the right UEFI/EFI driver .efi files to load the Install macOS dmg/img images, and therefore, boot the `boot.efi` located in `/System/Library/CoreServices/` in it.

**Comment by ElectroYeti** [comment]
Was (is)  there a way to boot it over opencore? Opencore is on fat32 drives, provides apfs drivers inbuild and hfs+ as well if it is correct configured.

**Comment by TheAlienDrew** [comment]
Ah I see what you mean, yeah OpenCore might be another good option to look source code https://github.com/dortania/OpenCore-Install-Guide

Never heard of it until now though, but could also be promising.

**Comment by ElectroYeti** [comment]
Yeah. But the apfs drivers seem to be in the binary compiled in the meantime. But there ARE apfs efi drivers in different releases, open source and the one from apple, with all its benefits and disadvantages. Some e.g. have a lot of logging in it ...

**Comment by ElectroYeti** [comment]
https://github.com/acidanthera/OcBinaryData/tree/master/Drivers but there are no apfs drivers, only for hpfs+ and exfat


Dont know if it is legal to post it as attachement , but in open core 0.56 there is in the Drivers Dircetory the 'Efi File ApfsDriverLoader.efi . Its one of the latest versions. 

Perhaps you could make it possible to chainload  
Ventoy - OpenCore - MacOS ?

**Comment by TheAlienDrew** [comment]
Well the Install macOS dmg files are normally created as HFS+ (Journaled), not APFS, at least that's how I've been creating mine. So I don't think we

**Comment by TheAlienDrew** [comment]
Also @ElectroYeti, the reason why there doesn't appear to be APFS drivers in there, is because it's already [build into OpenCore](https://www.reddit.com/r/hackintosh/comments/jnb8a4/is_hfsplusefi_no_longer_necessary_its_not_in_the/)

**Comment by TheAlienDrew** [comment]
But looks like there is an apfs driver located at https://github.com/allxiao/apfs.efi

**Comment by JonnyTech** [comment]
Has any progress been made with this? I am willing to test any suggestions.
